### PR TITLE
Add FastBoot support

### PR DESCRIPTION
**Pull request by SergeAstapov**
#41 has been stale and there were some controversial changes leading to discussions.
I've closed #41 and opening this PR to finally make this happen.

@fivetanley @jasonmit could you please take a look in this PR? Want to push this over the line and release a new version

### CODE REVIEW

**Comment by SergeAstapov** [comment]
tests fail with Node.js 4, created follow up #51

File: index.js [hunk 1]
```diff
@@ -86,6 +66,56 @@ module.exports = {
...
+   * Users may list in same keys both in `clientAllowedKeys` and
+   * `fastbootAllowedKeys`
+   *
+   * @returns {Object}
+   */
+  fastbootConfigTree() {
+    let loadedConfig = this._envConfig || {};
+    let fastbootAllowedKeys = this._config.fastbootAllowedKeys || [];
+
+    let config = fastbootAllowedKeys.reduce((accumulator, key) => {
```
Comment (jasonmit) [inline index.js]: Could be DRY'd up since this is very similar to the above reducer

**Comment by jasonmit** [comment]
I'd be fine with dropped Node 4 support or lowering fs-extra to where it's compat.

File: index.js (hunk 1 above)
Comment (SergeAstapov) [reply index.js]: yes, makes sense to me - PR has been updated

**Pull request merged**